common/djangoapps/dark_lang/views.py

```diff
@@ -0,0 +1,145 @@
+        """
+        context = {
+            'disable_courseware_js': True,
+            'uses_pattern_library': True
+        }
+        result = self.process_darklang_request(request)
+        if result is not None:
+            context.update({'form_submit_message': result})
+        return render_to_response(self.template_name, context)
+
+    def process_darklang_request(self, request):
+        """
+        Prevent user from requesting un-released languages
+        """
+        if not DarkLangConfig.current().enabled:
+            return None
+
+        if 'reset' in request.POST:
+            # Reset and clear the language preference
+            return self._clear_preview_language(request)
+        if 'set_language' in request.POST:
+            # Set the Preview Language
+            return self._set_preview_language(request)
+
+    def _set_preview_language(self, request):
+        """
+        Attempt to set the Preview language
+
+        Arguments:
+            request (Request): The incoming Django Request
+
+        Returns:
+            str: The response message to be presented
+        """
+        if LANGUAGE_INPUT_FIELD not in request.POST:
+            return _('Language code not provided')
+
+        preview_lang = request.POST[LANGUAGE_INPUT_FIELD]
+
+        if preview_lang == '':
+            return _('Language code not provided')
+
+        auth_user = request.user.is_authenticated()
+
+        # Set the session key to the requested preview lang
+        request.session[LANGUAGE_SESSION_KEY] = preview_lang
+
+        # Make sure that we set the requested preview lang as the dark lang preference for the
+        # user, so that the lang_pref middleware doesn't clobber away the dark lang preview.
+        if auth_user:
+            set_user_preference(request.user, DARK_LANGUAGE_KEY, preview_lang)
+        return _('Language set to language code: {}').format(preview_lang)
+
+    def _clear_preview_language(self, request):
+        """
+        Clears the dark language preview
+
+        Arguments:
+            request (Request): The incoming Django Request
+
+        Returns:
+            str: The response message to be presented
+        """
+        auth_user = request.user.is_authenticated()
+
+        # delete the session language key (if one is set)
+        if LANGUAGE_SESSION_KEY in request.session:
+            del request.session[LANGUAGE_SESSION_KEY]
+
+        user_pref = ''
+        if auth_user:
+            # Reset user's dark lang preference to null
+            delete_user_preference(request.user, DARK_LANGUAGE_KEY)
+            # Get & set user's preferred language
+            user_pref = get_user_preference(request.user, LANGUAGE_KEY)
+            if user_pref:
+                request.session[LANGUAGE_SESSION_KEY] = user_pref
+        if user_pref is None:
+            return _('Language reset user default')
+        return _('Language reset to default language code: {}').format(user_pref)
```

Comment:
Same note about named placeholder. And in this case, it isn't the default language is it?